lib/wickedbase.pm

```diff
@@ -117,6 +119,104 @@ sub assert_wicked_state {
     $self->ping_with_timeout(ip => $args{ping_ip}) if $args{ping_ip};
 }
 
+sub valgrind_get_services {
+    my $self = shift;
+    my $val = get_var('WICKED_VALGRIND');
+
+    return if (!defined($val) || !$val);
+    my @all = qw(wickedd-auto4
+      wickedd-dhcp6
+      wickedd
+      wickedd-dhcp4
+      wickedd-nanny
+      wicked);
+    my @tmp;
+    my @enable;
+    if (index($val, ",")) {
+        @tmp = split(/,/, $val);
+    } else {
+        @tmp = $val;
+    }
+    foreach my $v (@tmp) {
+        $v =~ s/\.service$//;
+        if (grep { /^$v$/ } @all) {
+            push @enable, $v;
+        } elsif ($v =~ /^1|all$/i) {
+            push @enable, @all;
+        } else {
+            record_info('WARNING', "Unknown value for WICKED_VALGRIND $v", result => 'softfail');
+        }
+    }
+    @enable = uniq @enable;
+
+    return @enable;
+}
+
+sub valgrind_cmd {
+    my ($self, $service) = @_;
+
+    my $valgrind_cmd = get_var('WICKED_VALGRIND_CMD', '/usr/bin/valgrind --tool=memcheck --leak-check=yes');
+    if ($service) {
+        my $logfile = "/var/log/valgrind_$service.log";
+        $valgrind_cmd = "$valgrind_cmd --log-file=$logfile";
+    }
+    return $valgrind_cmd;
+}
+
+sub valgrind_enable {
+    my $self = shift;
+    my $valgrind_cmd = get_var('WICKED_VALGRIND_CMD', '/usr/bin/valgrind --tool=memcheck --leak-check=yes');
+
+    my @services = $self->valgrind_get_services();
+    return 0 if (!@services);
+
+    record_info("valgrind enable", "services: @services\ncommand: $valgrind_cmd");
+
+    foreach my $service (@services) {
+        my $service_file = "/etc/systemd/system/$service.service";
+
+        assert_script_run("systemctl cat $service > $service_file");
+        assert_script_run(sprintf(q(sed -i -E 's@^(ExecStart=)(.*)$@\1%s \2@' '%s'),
+                $self->valgrind_cmd($service), $service_file));
+
+        record_info("$service.service", script_output("cat $service_file"));
+    }
+
+    assert_script_run('systemctl daemon-reload');
+
+    return 1;
+}
+
+sub valgrind_prerun {
+    my $self = shift;
+
+    my @services = $self->valgrind_get_services();
+    foreach my $service (@services) {
+        my $logfile = "/var/log/valgrind_$service.log";
+
+        assert_script_run("rm -f $logfile");
+    }
+}
+
+sub valgrind_postrun {
+    my $self = shift;
+
+    my @services = $self->valgrind_get_services();
+    foreach my $service (@services) {
+        my $logfile = "/var/log/valgrind_$service.log";
```

Comment:
haven't found upload of this log files anywhere . I think we should do this

Reply:
I didn't upload it, as I put it as `record_info()` if there is something relevant in it: http://openqa.wicked.suse.de/tests/136483#step/before_test/44

I think upload is to much here.

Reply:
up to you here , but from my personal experience very often in such cases when you have "something relevant" you want to see more context and get info what was happening before / after this relevant line  , so maybe we can upload full log when something relevant was found ? 

But don't want to insists so I am resolving this thread keeping this on your good will

Reply:
But that is what you get in the `record_info()` it contain the full file not only the regex match.
So if we upload the file, we do not get more info, just from a other place...

Reply:
ah ok